### PR TITLE
Some missing bidiag tests

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: stdlib/LinearAlgebra/test/bidiag.jl [hunk 1]
```diff
@@ -297,9 +300,14 @@ Random.seed!(1)
             # test pass-through of mul! for AbstractTriangular*Bidiagonal
             Tri = UpperTriangular(diagm(1 => T.ev))
             @test Array(Tri*T) ≈ Array(Tri)*Array(T)
+            # test mul! for Diagonal*Bidiagonal
+            C = Matrix{elty}(undef, n, n)
+            Dia = Diagonal(T.dv)
+            @test mul!(C, Dia, T) ≈ Array(Dia)*Array(T)
```
Comment (dkarrasch) [inline stdlib/LinearAlgebra/test/bidiag.jl]: What about testing also `mul!(C,::Bidiagonal,::Diagonal)`, or is there no such method?

Comment (kshyatt) [reply stdlib/LinearAlgebra/test/bidiag.jl]: There isn't one in `bidiag.jl`, at least...

Comment (dkarrasch) [reply stdlib/LinearAlgebra/test/bidiag.jl]: Hm, there is something related going on in #31889, maybe we should return to this once that specialized functionality exists.

**Comment by kshyatt** [comment]
Is this mergeable?

**Review by dkarrasch — APPROVED** [review]
I'd say yes. #31889 is still in the line, so we'll need to take care of my comment there.

**Pull request merged**